### PR TITLE
Add alert component

### PR DESCRIPTION
**Pull request by azizullahsaeidi**
1. Create alert message component
2. Add this component on `update user profile `, `Create new resource` and `Add subscribe`
3. Fixed `resource list content` css style, changed flex to grade

### CODE REVIEW

File: resources/views/users/view_user.blade.php [hunk 1]
```diff
@@ -1,112 +1,128 @@
...
+                            <div class="flex-5">
+                                @if ($user->status == 0)
+                                    <span class="user-state bg-red-300">Disabled
+                                        <span class="fa fa-times-circle"></span></span>
+                                @else
+                                    <span class="user-state bg-lightseagreen">Active
+                                        <span class="fa fa-check-circle"></span></span>
+                                @endif
+                            </div>
+                        </div>
```
Comment (uyscuti-wiki) [inline resources/views/users/view_user.blade.php:100]: If `status` is anything other than `1`, we can reasonably assume that they won't (or shouldn't rather) be able to login, making the `status` display redundant here. Let's remove this and check if `status` is `1` before allowing them to view this page. 

If it's not `1`, let's log them out.

**Review by uyscuti-wiki — CHANGES_REQUESTED** [review]
Excellent work adding session alerts, Aziz. I have requested some changes. This should be good to go once it is all sorted out. 

While we are on this, let's also make it a practice of adding tests for all the new features we add going forward. Let's start with this PR.

File: resources/views/subscribe/index.blade.php [hunk 2]
```diff
@@ -10,83 +10,90 @@
...
+                                </small>
 
-                    </form>
+                            </form>
+                        </div>
+                    @else
+                        {{-- Already Subscribed --}}
+                        <header>
+                            <h2 class="text-center">@lang('Thank you')</h2>
+                            <p class="text-center">@lang('You subscribed our newsletter.')</p>
```
[the remark this answers is not in the document]
Comment (azizullahsaeidi) [reply resources/views/subscribe/index.blade.php]: Great idea. Thanks

File: resources/assets/css/ddl.css [hunk 3]
```diff
@@ -624,23 +624,19 @@ aside .sub-subject-show {
...
     display: flex;
     justify-content: flex-start;
-    align-items: flex-start;
     align-content: flex-start;
 }
 
 .resource-information-section .resource-article:hover {
     cursor: pointer;
-    opacity: 0.7;
 }
```
[the remark this answers is not in the document]
Comment (azizullahsaeidi) [reply resources/assets/css/ddl.css:632]: Yes you are right, we should remove it.

File: resources/views/layouts/search.blade.php [hunk 4]
```diff
@@ -1,12 +1,27 @@
...
-        @csrf
-        <label for="search">
-            <input type="search" name="search" class="form-search form-control" value="{{ session('search') }}" placeholder="@lang('SEARCH OUR GROWING LIBRARY!')" autofocus>
-        </label>
-        <input type="submit" class="search-button" value="@lang('Go')">
-    </form>
+    <div class="main-section search-section">
+        @if (session()->has('alert'))
+            <x-alert :message="Session::get('alert.message')" :level="Session::get('alert.level')" />
+        @endif
```
[the remark this answers is not in the document]
Comment (azizullahsaeidi) [reply resources/views/layouts/search.blade.php:5]: I have added z-index

[the remark this answers is not in the document]
Comment (uyscuti-wiki) [reply resources/views/layouts/search.blade.php:5]: I still believe the placement is problematic. A block-style display that wouldn't overlap on existing content is the way to go. 

image

[the remark this answers is not in the document]
Comment (uyscuti-wiki) [reply resources/views/layouts/search.blade.php:5]: Centred on the page if possible.

[the remark this answers is not in the document]
Comment (uyscuti-wiki) [reply resources/views/layouts/search.blade.php:5]: image

**Review by uyscuti-wiki — CHANGES_REQUESTED** [review]
Thank you Aziz. I am still seeing the alert overlaid upon an existing element. We discussed about how different keys and strings affect the code readability. Please also implement it. Should be good to go then.

File: lang/en.json [hunk 5]
```diff
@@ -2,5 +2,6 @@
     "Receive our occasional newsletters and announcements (your email will be shared with MailChimp).":"Receive our occasional newsletters and announcements (your email will be shared with <a href='<a href='https://mailchimp.com/legal/'>MailChimp</a>).",
     "Your email will be shared with MailChimp.": "Your email will be shared with MailChimp. <br> <a href='https://mailchimp.com/legal/'> Their Privacy Policy",
     "You'll receive our monthly newsletter with our latest updates, highlights, top resources, educational content and other relevant news.": "You'll receive our monthly newsletter with<br> our latest updates, highlights, top resources,<br> educational content and other relevant news.",
-    "If you registered using a phone number, please contact us.": "If you registered using a phone number, please <a href='/contact-us'>contact us</a>."
+    "If you registered using a phone number, please contact us.": "If you registered using a phone number, please <a href='/contact-us'>contact us</a>.",
+    "Error updating data. Please contact us.": "Error updating data. Please <a href='/contact-us'>contact us</a>."
```
[the remark this answers is not in the document]
Comment (azizullahsaeidi) [reply lang/en.json]: Removed this code

File: app/Http/Controllers/UserController.php [hunk 6]
```diff
@@ -119,10 +120,19 @@ public function updateProfile(Request $request): RedirectResponse
...
+                'message' => __('Your data successfully updated.'),
+                'level' => 'success'
+            ]);
+
+            return redirect(URL('user/profile'));
         }
+        Session::flash('alert', [
+            'message' => __('Sorry! your data was not updated.'),
+            'level' => 'danger'
+        ]);
```
[the remark this answers is not in the document]
Comment (azizullahsaeidi) [reply app/Http/Controllers/UserController.php]: Removed this message

File: resources/views/layouts/search.blade.php (hunk 4 above)
[the remark this answers is not in the document]
Comment (azizullahsaeidi) [reply resources/views/layouts/search.blade.php]: Solved this issue

File: resources/views/subscribe/index.blade.php [hunk 7]
```diff
@@ -10,83 +10,90 @@
...
+                                </div>
 
-                        {{-- Mailchimp --}}
-                        <p style="color:gray; font-size: 12px;">
-                            @lang('Your email will be shared with MailChimp.')
-                        </p>
+                                {{-- Mailchimp --}}
+                                <small style="color:gray">
+                                    @lang('Your email will be shared with MailChimp. their privacy policy')
+                                </small>
```
[the remark this answers is not in the document]
Comment (azizullahsaeidi) [reply resources/views/subscribe/index.blade.php]: image

File: resources/views/subscribe/index.blade.php (hunk 2 above)
[the remark this answers is not in the document]
Comment (azizullahsaeidi) [reply resources/views/subscribe/index.blade.php]: Removed this section instead we redirect to home page

**Comment by azizullahsaeidi** [comment]
> Please move the alert on the user page to the top. image and I'll merge this.

It is displaying on the top the page.
image

**Comment by uyscuti-wiki** [comment]
Not on the header, Aziz. Below 'User Details..' if possible.

**Comment by azizullahsaeidi** [comment]
> Not on the header, Aziz. Below 'User Details..' if possible.

I added this component only one time, If I move it below the `User Details`, then we need to duplicate the code.

**Comment by uyscuti-wiki** [comment]
How about below the header then? I just don't like the fact that it's overlapping other elements.

**Review by uyscuti-wiki — APPROVED** [review]
Thank you Aziz. I appreciate your patience during the review.

**Pull request merged**